x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/data/BlockSerializationTests.java

```diff
@@ -125,37 +128,47 @@ public void testFilterBytesRefBlock() throws IOException {
 
     private void assertFilterBlock(Block origBlock) throws IOException {
         assertThat(origBlock.getPositionCount(), is(1));
-        Block deserBlock = serializeDeserializeBlock(origBlock);
-        EqualsHashCodeTestUtils.checkEqualsAndHashCode(origBlock, unused -> deserBlock);
-        assertThat(deserBlock.getPositionCount(), is(1));
+        try (Block deserBlock = serializeDeserializeBlock(origBlock)) {
+            EqualsHashCodeTestUtils.checkEqualsAndHashCode(origBlock, unused -> deserBlock);
+            assertThat(deserBlock.getPositionCount(), is(1));
+        }
     }
 
     public void testConstantNullBlock() throws IOException {
         Block origBlock = new ConstantNullBlock(randomIntBetween(1, 8192));
-        Block deserBlock = serializeDeserializeBlock(origBlock);
-        EqualsHashCodeTestUtils.checkEqualsAndHashCode(origBlock, unused -> deserBlock);
+        try (Block deserBlock = serializeDeserializeBlock(origBlock)) {
+            EqualsHashCodeTestUtils.checkEqualsAndHashCode(origBlock, unused -> deserBlock);
+        }
     }
 
     // TODO: more types, grouping, etc...
-    public void testAggregatorStateBlock() throws IOException {
+    public void testAggregatorStateBlock() {  // TODO we don't have an aggregator state block any more. Do we need this?
```

Comment:
We no longer have an aggStateBlock, but I do like this test case. Maybe just rename? as it simulates an agg intermediate state output.